test/runtests.jl

```diff
@@ -1,3 +1,8 @@
+@static if get(ENV, "JULIA_GITHUB_ACTIONS_CI", "OFF") == "ON" && Sys.isapple()
+	import Libdl: DL_LOAD_PATH
+	push!(DL_LOAD_PATH, ENV["JULIA_VULKAN_SDK_SEARCH_PATH"])
+end
```

Comment:
Why this is needed on macOS?

Reply:
That's because `libvulkan.dylib` is not on the standard search path so you need to add one. The code that was in `__init__` handled it by adding the `JULIA_VULKAN_SDK_SEARCH_PATH` to the search path for `Libdl` as a fallback before opening the library. While it retrieved a valid handle with `dlopen`, it was not linked in any way to the `libvulkan` variable that is used for the `ccall`s. Therefore I believe any Vulkan commands would have errored, even if `__init__` succeeded. I can't say for sure, because there are no tests on MacOS (everything is skipped). But basically now we throw an error at init time if the `libvulkan` is not valid for `ccall`.

Reply:
I forgot why I was setting those environment variables like this, looks like I was doing it wrong. I just checked those environment variable on my machine, `export DYLD_LIBRARY_PATH` is being used instead of `JULIA_VULKAN_SDK_SEARCH_PATH`.

Reply:
Then I think we should remove the above altogether and just set this environment variable.

Reply:
Did you do anything more? Just adding the library folder to `DYLD_LIBRARY_PATH` does not seem to work.

Reply:
Oops! I also added `$VULKAN_SDK/bin` to `PATH`.
```
export VULKAN_SDK=/Users/gnimuc/vulkan/macOS
export PATH="$PATH:$VULKAN_SDK/bin"
export DYLD_LIBRARY_PATH="$VULKAN_SDK/lib:${DYLD_LIBRARY_PATH:-}"
export VK_LAYER_PATH="$VULKAN_SDK/share/vulkan/explicit_layer.d"
export VK_ICD_FILENAMES="$VULKAN_SDK/share/vulkan/icd.d/MoltenVK_icd.json"
```

Reply:
But I don't think that matters since `$VULKAN_SDK/bin` only contains binaries.
```
➜  vulkan ls macOS/bin 
dxc               spirv-as          spirv-lesspipe.sh spirv-remap
dxc-3.7           spirv-cfg         spirv-link        spirv-val
glslangValidator  spirv-cross       spirv-opt         vkvia
glslc             spirv-dis         spirv-reduce      vulkaninfo
```

Reply:
Hmm you're right. I just pushed the same setup as yours, we'll see.

Reply:
No success indeed. I reverted to use DL_LOAD_PATH like before. If you can come up with a solution, I'm glad to test it, otherwise I think it's ok for CI to just use this hack.

Reply:
Agreed. I don't have time to dive into this for now, will file an issue to track this. I also checked `Libdl.DL_LOAD_PATH`, it doesn't pick the `DYLD_LIBRARY_PATH` environment variable on my machine, but both VulkanCore and GLFW work fine.

Reply:
I also checked the `Libdl.DL_LOAD_PATH` variable on my machine, it doesn't pick up `DYLD_LIBRARY_PATH`, but both VulkanCore and GLFW works fine. 

```
julia> Libdl.DL_LOAD_PATH
2-element Array{String,1}:
 "@loader_path/julia"
 "@loader_path"
```

I don't have time to dive into this for now, will file an issue to keep tracking this.